### PR TITLE
Input validation redesign

### PR DESCRIPTION
**Pull request by zpevma**
- input validation redesigned, we can now tolerate specific errors (check for substring in error message)
 - determine cvss v 3.x version from `ScoreSetV3` element namespace

### CODE REVIEW

File: cvrf2csaf/cvrf2csaf.py [hunk 1]
```diff
@@ -34,6 +34,9 @@ class DocumentHandler:
     5. Combining it to the final JSON and writing the result to a file
     """
 
+    TOLERATED_ERRORS_SUBSTR = ["}ScoreSetV3': This element is not expected. Expected is one of ( {http://docs.oasis-open.org/csaf/ns/csaf-cvrf/v1.2/vuln}ScoreSetV2, {http://docs.oasis-open.org/csaf/ns/csaf-cvrf/v1.2/vuln}ScoreSetV3 ).",
```
Comment (sthagen) [inline cvrf2csaf/cvrf2csaf.py:37]: Nit: `"}ScoreSetV3':`  &rightarrow; `"ScoreSetV3:`?

File: cvrf2csaf/cvrf2csaf.py [hunk 2]
```diff
@@ -34,6 +34,9 @@ class DocumentHandler:
...
+                        "ScoreSetV3': This element is not expected. Expected is ( {http://docs.oasis-open.org/csaf/ns/csaf-cvrf/v1.2/vuln}ScoreSetV3 )."]
```
Comment (sthagen) [inline cvrf2csaf/cvrf2csaf.py:38]: Nit: `"ScoreSetV3':` &rightarrow; `"ScoreSetV3:`

File: cvrf2csaf/cvrf2csaf.py (hunk 1 above)
Comment (tschmidtb51) [reply cvrf2csaf/cvrf2csaf.py:37]: I guess it is to see that it is a namespace error...

Comment (tschmidtb51) [reply cvrf2csaf/cvrf2csaf.py:37]: @sthagen The original error message according to https://github.com/csaf-tools/CVRF-CSAF-Converter/pull/56#discussion_r808070742 was:

`Element '{https://www.first.org/cvss/cvss-v3.1.xsd}ScoreSetV3': This element is not expected. Expected is one of ( {http://docs.oasis-open.org/csaf/ns/csaf-cvrf/v1.2/vuln}ScoreSetV2, {http://docs.oasis-open.org/csaf/ns/csaf-cvrf/v1.2/vuln}ScoreSetV3 )`

So this looks okay to me.

File: cvrf2csaf/cvrf2csaf.py (hunk 2 above)
Comment (tschmidtb51) [reply cvrf2csaf/cvrf2csaf.py:38]: @zpevma In which case does that error message occur?

**Review by tschmidtb51 — APPROVED** [review]
Not tested (I trust for now that you did that). But the code looks good to me.

File: cvrf2csaf/cvrf2csaf.py (hunk 2 above)
Comment (tschmidtb51) [inline cvrf2csaf/cvrf2csaf.py:38]: @zpevma aren't we currently missing the error string for #57 here? Or will that be a separate PR?

File: cvrf2csaf/cvrf2csaf.py (hunk 1 above)
Comment (zpevma) [reply cvrf2csaf/cvrf2csaf.py:37]: @tschmidtb51 exactly, I just picked the longest fixed substring containing this particular error. I will unify both substrings (add '}' to the second one).

File: cvrf2csaf/cvrf2csaf.py (hunk 2 above)
Comment (zpevma) [reply cvrf2csaf/cvrf2csaf.py:38]: for example this:
```
<CVSSScoreSets>
      <ScoreSetV2>
        <BaseScoreV2>7.5</BaseScoreV2>
        <VectorV2>AV:A/AC:M/Au:N/C:N/I:C/A:P</VectorV2>
      </ScoreSetV2>
      <ScoreSetV3>
        <BaseScoreV3>7.5</BaseScoreV3>
        <VectorV3>CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H</VectorV3>
      </ScoreSetV3>
      <ScoreSetV3 xmlns="https://www.first.org/cvss/cvss-v3.0.xsd">
        <BaseScoreV3>7.5</BaseScoreV3>
        <VectorV3>CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H</VectorV3>
      </ScoreSetV3>
</CVSSScoreSets>
```
`ScoreSetV3` element with namespace appears after `ScoreSetV3` without namespace. Then the error message is different from previous one.

Comment (zpevma) [reply cvrf2csaf/cvrf2csaf.py:38]: Yes, there will be one or two more PR.

File: cvrf2csaf/cvrf2csaf.py (hunk 1 above)
Comment (sthagen) [reply cvrf2csaf/cvrf2csaf.py:37]: 🙈 I was assuming (wrongly) that this was an output string and not a substring. Sorry for the noise.

**Review by sustefil — APPROVED** [review]
LGTM
